### PR TITLE
Add KB - VM Scheduling in Harvester

### PR DESCRIPTION
**Pull request by FrankYang0529**
Related issue: https://github.com/harvester/harvester/issues/1654

### CODE REVIEW

File: kb/2022-02-24/vm-scheduling.md [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+            matchExpressions:
+            - key: harvesterhci.io/creator
+              operator: Exists
+          topologyKey: kubernetes.io/hostname
+        weight: 100
+```
+
+## How to see scores in kube-scheduler?
+
+kube-scheduler is deployed as a static pod in Harvester. The file is under `/var/lib/rancher/rke2/agent/pod-manifests/kube-scheduler.yaml` in each Management Node. We can add `- --v=10` to the kube-scheduler container to show score logs.
```
Comment (nunix) [inline kb/2022-02-24/vm-scheduling.md:375]:
```suggestion
`kube-scheduler` is deployed as a static pod in Harvester. The file is under `/var/lib/rancher/rke2/agent/pod-manifests/kube-scheduler.yaml` in each Management Node. We can add `- --v=10` to the `kube-scheduler` container to show score logs.
```

**Review by nunix — COMMENTED** [review]
Rephrased to active voice and moved some parts to aligned with the examples/tables.

File: kb/2022-02-24/vm-scheduling.md [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+| NodeAffinity                    |                0 |                0 |                0 |
+| NodePreferAvoidPods             |          1000000 |          1000000 |          1000000 |
+| PodTopologySpread               |              200 |              200 |              200 |
+| TaintToleration                 |              100 |              100 |              100 |
+| NodeResourcesBalancedAllocation |                0 |               45 |               46 |
+| Total                           |          1000358 |          1000433 |          1000437 |
+
+## Why VMs are distributed unevenly with overcommit?
+
+With overcommit, Harvester modifies request resources. By default, overcommit configuration is `{"cpu": 1600, "memory": 150, "storage":  200}`. If we request a VM with 1 CPU and 1G RAM, its `resources.requests.cpu` will become `62m`. It's because 1 CPU is equal to 1000m CPU and overcommit configuration `"cpu": 1600` means make CPU 16x smaller. `1000m * 100 / 1600 = 62m`. Now, we can see how overcommitting influences kube-scheduler scores.
```
[the remark this answers is not in the document]
Comment (FrankYang0529) [reply kb/2022-02-24/vm-scheduling.md]: Updated it. Thank you.

File: kb/2022-02-24/vm-scheduling.md [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+"Calculated node's final score for pod" pod="default/virt-launcher-vm2-without-overcommit-mf5vk" node="harvester-node-0" score=1000358
+"Calculated node's final score for pod" pod="default/virt-launcher-vm2-without-overcommit-mf5vk" node="harvester-node-1" score=1000433
+"Calculated node's final score for pod" pod="default/virt-launcher-vm2-without-overcommit-mf5vk" node="harvester-node-2" score=1000382
+
+AssumePodVolumes for pod "default/virt-launcher-vm2-without-overcommit-mf5vk", node "harvester-node-1"
+AssumePodVolumes for pod "default/virt-launcher-vm2-without-overcommit-mf5vk", node "harvester-node-1": all PVCs bound and nothing to do
+"Attempting to bind pod to node" pod="default/virt-launcher-vm2-without-overcommit-mf5vk" node="harvester-node-1"
+```
+</details>
+
```
[the remark this answers is not in the document]
Comment (FrankYang0529) [reply kb/2022-02-24/vm-scheduling.md:356]: Update content for this paragraph. Thank you.

**Comment by guangbochen** [comment]
can u please comment out the following code to enable the KB page and run `$ yarn install & yarn start` to validate it locally, thanks?
https://github.com/harvester/harvesterhci.io/blob/04ac39da2634378579e9312c32ff2793b820db09/docusaurus.config.js#L41-L47,

**Comment by FrankYang0529** [comment]
> can u please comment out the following code to enable the KB page and run `$ yarn install & yarn start` to validate it locally, thanks?

Yes, I tried it and it works fine. I can click links, and all content show as expected.

**Review by nunix — COMMENTED** [review]
Create a note for the suffix `m`.
Changed formatting for Table X back to italic as it provides a better readability.

File: kb/2022-03-02/vm-scheduling.md [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+hide_table_of_contents: false
+---
+
+# VM Scheduling in Harvester
+
+## How does Harvester schedule a VM?
+
+Harvester doesn't directly schedule a VM in Kubernetes, it relies on [KubeVirt](http://kubevirt.io/) to create the custom resource `VirtualMachine`. When the request to create a new VM is sent, a `VirtualMachineInstance` object is created and it creates the corresponding `Pod`.
+
+ The whole VM creation processt leverages `kube-scheduler`, which allows Harvester to use `nodeSelector`, `affinity`, and resources request/limitation to influence where a VM will be deployed.
```
Comment (nunix) [inline kb/2022-03-02/vm-scheduling.md:20]:
```suggestion
The whole VM creation processt leverages `kube-scheduler`, which allows Harvester to use `nodeSelector`, `affinity`, and resources request/limitation to influence where a VM will be deployed.
```

File: kb/2022-03-07/vm-scheduling.md [hunk 5]
```diff
@@ -0,0 +1,416 @@
...
+## Why VMs are distributed unevenly with overcommit?
+
+With resources overcommit, Harvester modifies the resources request. By default, the `overcommit` configuration is `{"cpu": 1600, "memory": 150, "storage":  200}`. This means that if we request a VM with 1 CPU and 1G RAM, its `resources.requests.cpu` will become `62m`. 
+
+!!! note
+    The unit suffix `m` stands for "thousandth of a core."
+
+To explain it, let's take the case of CPU overcommit. The default value of 1 CPU is equal to 1000m CPU, and with the default overcommit configuration of `"cpu": 1600`, the CPU resource will be 16x smaller. Here is the calculation: `1000m * 100 / 1600 = 62m`.
+
+Now, we can see how overcommitting influences kube-scheduler scores.
```
Comment (nunix) [inline kb/2022-03-07/vm-scheduling.md:109]:
```suggestion
Now, we can see how overcommitting influences `kube-scheduler` scores.
```

File: kb/2022-03-07/vm-scheduling.md [hunk 6]
```diff
@@ -0,0 +1,416 @@
...
+
+## How does kube-scheduler decide where to deploy a VM?
+
+First, `kube-scheduler` finds Nodes available to run a pod. After that, `kube-scheduler` scores each available Node by a list of [plugins](https://github.com/kubernetes/kubernetes/tree/v1.22.7/pkg/scheduler/framework/plugins) like [ImageLocality](https://github.com/kubernetes/kubernetes/blob/v1.22.7/pkg/scheduler/framework/plugins/imagelocality/image_locality.go), [InterPodAffinity](https://github.com/kubernetes/kubernetes/tree/v1.22.7/pkg/scheduler/framework/plugins/interpodaffinity), [NodeAffinity](https://github.com/kubernetes/kubernetes/tree/v1.22.7/pkg/scheduler/framework/plugins/nodeaffinity), etc. 
+
+Finally, `kube-scheduler` calculates the scores from the plugins results for each Node, and select the Node with the highest score to deploy the Pod.
+
+For example, let's say  we have a three node Harvester cluster with 6 cores CPU and 16G RAM each, and we want to deploy a VM with 1 CPU and 1G RAM (without resources overcommit).
```
Comment (nunix) [inline kb/2022-03-07/vm-scheduling.md:28]:
```suggestion
For example, let's say  we have a three nodes Harvester cluster with 6 cores CPU and 16G RAM each, and we want to deploy a VM with 1 CPU and 1G RAM (without resources overcommit). 
```

File: kb/2022-03-07/vm-scheduling.md [hunk 7]
```diff
@@ -0,0 +1,416 @@
...
+With resources overcommit, Harvester modifies the resources request. By default, the `overcommit` configuration is `{"cpu": 1600, "memory": 150, "storage":  200}`. This means that if we request a VM with 1 CPU and 1G RAM, its `resources.requests.cpu` will become `62m`. 
+
+!!! note
+    The unit suffix `m` stands for "thousandth of a core."
+
+To explain it, let's take the case of CPU overcommit. The default value of 1 CPU is equal to 1000m CPU, and with the default overcommit configuration of `"cpu": 1600`, the CPU resource will be 16x smaller. Here is the calculation: `1000m * 100 / 1600 = 62m`.
+
+Now, we can see how overcommitting influences kube-scheduler scores.
+
+In this example, we use a three node Harvester cluster with 6 cores and 16G RAM each. We will deploy two VMs with 1 CPU and 1G RAM, and we will compare the scores for both cases of "with-overcommit" and "without-overcommit" resources.
```
Comment (nunix) [inline kb/2022-03-07/vm-scheduling.md:111]:
```suggestion
In this example, we use a three nodes Harvester cluster with 6 cores and 16G RAM each. We will deploy two VMs with 1 CPU and 1G RAM, and we will compare the scores for both cases of "with-overcommit" and "without-overcommit" resources. 
```

**Review by nunix — COMMENTED** [review]
few last corrections

**Review by nunix — APPROVED** [review]

**Review by guangbochen — APPROVED** [review]

**Pull request merged**